app/controllers/home_controller.rb

```diff
@@ -30,7 +30,20 @@ def landing
 
   def index
     if current_user.allowed_feature_list.include?("data_discovery")
-      redirect_to my_data_path
+      project_id = params[:project_id]
+      project = Project.find(params[:project_id])
+
+      redirect_to my_data_path if project.blank?
+
+      if current_user.owns_project?(project.id)
+        redirect_to action: "my_data", project_id: project_id
```

Comment:
I don't see `my_data` action as a method in `HomeController`... what am I missing about rails routing?

Reply:
The action does nothing it goes directly to view where it adds the React component. In that case you do not need to define it.